log/logger_multiwriter.go

```diff
@@ -40,42 +43,73 @@ func (mw *multiWriterHolder) Remove(writer io.Writer) error {
 	return errWriterNotFound
 }
 
-// Write concurrent safe Write for each writer
-func (mw *multiWriterHolder) Write(p []byte) (int, error) {
-	type data struct {
-		n   int
-		err error
-	}
-
-	results := make(chan data, len(mw.writers))
-	mw.mu.RLock()
-	defer mw.mu.RUnlock()
-	for x := range mw.writers {
-		go func(w io.Writer, p []byte, ch chan<- data) {
-			n, err := w.Write(p)
-			if err != nil {
-				ch <- data{n, fmt.Errorf("%T %w", w, err)}
-				return
+func loggerWorker() {
```

Comment:
While I think its mostly implicit already, but since this function doesn't take in a `sync.Waitgroup` param and uses a global one, there is a risk of bad behaviour if this is called outside of the exact routine with the `workerWg.Done()`. I'm just thinking, perhaps add a comment to this function to make it explicit that this is to be called as a go routine